packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -348,5 +464,22 @@ class _CupertinoButtonBarPainter extends CustomPainter {
   }
 
   @override
-  bool shouldRepaint(_CupertinoButtonBarPainter other) => count != other.count;
+  bool shouldRepaint(_CupertinoHorizontalDividerPainter other) => count != other.count;
+}
+
+/// A CustomPainter to draw the divider lines.
+///
+/// Draws the cross-axis divider lines, used when the layout is vertical.
+class _CupertinoVerticalDividerPainter extends CustomPainter {
+  _CupertinoVerticalDividerPainter();
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final Paint paint = new Paint()
+      ..color = _kButtonDividerColor;
+    canvas.drawLine(const Offset(0.0, 0.0), new Offset(size.width, 0.0), paint);
+  }
+
+  @override
+  bool shouldRepaint(_CupertinoVerticalDividerPainter other) => true;
```

Comment:
This is essentially saying, CustomPaint instances might be created all the time while creating the widget tree. When a new one is created, does the existing RenderCustomPaint need to call paint again. Since this painter always paints the same way, you can return false here.

Reply:
Got it.